src/hooks/useHtmlPaste/index.ts

```diff
@@ -0,0 +1,129 @@
+    );
+
+    /**
+     * Paste the plaintext content into Composer.
+     *
+     * @param {ClipboardEvent} event
+     */
+    const handlePastePlainText = useCallback(
+        (event: ClipboardEvent) => {
+            const plainText = event.clipboardData?.getData('text/plain');
+            if (plainText) {
+                paste(plainText);
+            }
+        },
+        [paste],
+    );
+
+    const handlePaste = useCallback(
+        (event: ClipboardEvent) => {
+            if (!textInputRef.current) {
+                return;
+            }
+            const isVisible = typeof checkComposerVisibility === 'function' && checkComposerVisibility();
+            const isFocused = textInputRef.current?.isFocused();
+
+            if (!isFocused && !isVisible) {
+                return;
+            }
+
+            if (textInputRef.current !== event.target) {
+                // To make sure the text input does not capture paste events from other inputs, we check where the event originated
+                // If it did originate in another input, we return early to prevent the text input from handling the paste
+                const target = event.target as HTMLInputElement;
+                const isTargetInput = (target && target.nodeName === 'INPUT') || target.nodeName === 'TEXTAREA' || target.contentEditable === 'true';
+
+                if (isTargetInput) {
+                    return;
+                }
+
+                textInputRef.current?.focus();
+            }
+
+            event.preventDefault();
+
+            const types = event.clipboardData?.types;
+            const TEXT_HTML = 'text/html';
+
+            // If paste contains HTML
+            if (types && types.includes(TEXT_HTML)) {
+                const pastedHTML = event.clipboardData.getData(TEXT_HTML);
+
+                const domparser = new DOMParser();
+                const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML).images;
+
+                // Exclude parsing img tags in the HTML, as fetching the image via fetch triggers a connect-src Content-Security-Policy error.
+                if (embeddedImages.length > 0 && embeddedImages[0].src) {
+                    // If HTML has emoji, then treat this as plain text.
+                    if (embeddedImages[0].dataset && embeddedImages[0].dataset.stringifyType === 'emoji') {
+                        handlePastePlainText(event);
+                        return;
+                    }
+                }
+                handlePastedHTML(pastedHTML);
+                return;
+            }
+            handlePastePlainText(event);
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [handlePastedHTML, handlePastePlainText, checkComposerVisibility],
+    );
+
+    useEffect(() => {
+        // we need to handle listeners on navigation focus/blur if the component (like Composer) is not unmounting
+        // when navigating away to different screen (report)
```

Comment:
I would add an "or else" here to explain the consequences of not doing this. 

We are explaining what we are doing - but the side effect would be... handling a paste event on both inputs? How is it even possible if the paste method returned early when the composer is not visible. Don't really get that part tbh.

Reply:
```
```

Reply:
Please update the comment so that it explains _why_ we need to do this